clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -243,6 +247,124 @@ class AzureNetworkClient extends AzureBaseClient {
   }
 
   /**
+   * Retrieve an Azure Application Gateway for a give set of credentials, resource group and name
+   * @param resourceGroupName the name of the resource group to look into
+   * @param appGatewayName the of the application gateway
+   * @return a description object which represent an application gateway in Azure or null if the application gateway resource could not be retrieved
+   */
+  AzureAppGatewayDescription getAppGateway(String resourceGroupName, String appGatewayName) {
+    try {
+      def currentTime = System.currentTimeMillis()
+      def appGateway = executeOp({appGatewayOps.get(resourceGroupName, appGatewayName)})?.body
+      if (appGateway) {
+        def agItem = AzureAppGatewayDescription.getDescriptionForAppGateway(appGateway)
+        agItem.appName = AzureUtilities.getAppNameFromResourceId(appGateway.id)
+        agItem.tags = appGateway.tags
+        agItem.dnsName = getDnsNameForAppGateway(AzureUtilities.getResourceGroupNameFromResourceId(appGateway.id), appGateway.name)
+        agItem.lastReadTime = currentTime
+        return agItem
+      }
+    } catch (CloudException e) {
+      log.error("getAppGateway(${resourceGroupName},${appGatewayName}) -> Cloud Exception ", e)
+    } catch (Exception e) {
+      log.error("getAppGateway(${resourceGroupName},${appGatewayName}) -> Exception ", e)
+    }
```

Comment:
There is no need for two different 'catch' clauses. The output will show what kind of exception it actually is.